### DISCUSSION TITLE
Embed within another Phoenix Application

### DISCUSSION THREAD

**Discussion by mikehostetler** [zoonk/uneebee#175] [Ideas]
I'd love to be able to embed `uneebee` within another Phoenix project - rather then deploy it as a standalone system.

Any interest for this? Would love to understand if there is any other interest or demand from others.

**Comment by ceolinwill** [comment]
In theory, we could follow an approach similar to the [Phoenix LiveDashboard](https://github.com/phoenixframework/phoenix_live_view) to accomplish this but I'm unsure how to handle the data.

For example, how to handle users and authorization? I'm guessing you'd like your users to have access to it without having to create another account. But `uneebee` has its own `User` schema, so we'd need to find a way to conciliate both user schemas.

Another concern is handling migrations. At first sight, it seems a bit tricky to get migrations right for both the main and the embedded application.

I've seen another example of an [embedded application](https://elixirforum.com/t/is-there-a-good-way-to-embed-a-phoenix-app-to-another-phoenix-app/48328/1) but it doesn't use Ecto either.

If anyone has any suggestions on how to accomplish this, please let me know.

**Comment by mikehostetler** [comment]
Yeah, there's a few places to make the integration happen - here's what I'd look for:

1. Config
2. Router
3. Migrations (perhaps with a table name prefix?)
4. Templating (to match visual style)
5. Functionality Hooks

Some good examples I've seen on different pieces of this:

* https://github.com/mojotech/torch
* https://github.com/aesmail/kaffy
* https://hexdocs.pm/pow/README.html
* https://github.com/phoenixframework/phoenix_live_dashboard

Hope that helps!

**Reply by ceolinwill** [reply]
Thanks. I'll have a look into those.

A schema prefix would prevent conflicts but I think the main issue is syncing the UneeBee user with the main app's user to avoid having them to sign in/sign up to the embedded route as well.

I'll try to play with this idea on my spare time to see if I can create a proof of concept but I'm not sure yet.

**Reply by ceolinwill** [reply]
@mikehostetler Just an update: I didn't forget about this but there are edge cases I'd have to handle that could be a distraction and I'm afraid to end up spending too much time on this feature.

So, I'm not going to prioritize this for now to focus on the [current roadmap](https://github.com/orgs/zoonk/projects/11) because I need those things to start monetizing this project with a cloud offering.

But I'm keeping this open because I do think this is a nice feature to have eventually - just not something that I can tackle right now. I hope you understand.

**Reply by mikehostetler** [reply]
@wceolin No worries and thanks for the followup. You're building an awesome project and we'll be following along!

**Comment by codeanpeace** [comment]
Rather than embedding it within another Phoenix app, it'd be really compelling to be able to embed it into any other app via a custom element with something like [LiveState](https://hexdocs.pm/live_state). That would mean it could be used in anything from a Wordpress app, static site, or even another Phoenix app. For non-public resources, there'd need to be some setup so that the client side can receive a user specific public key of some sort.

**Reply by ceolinwill** [reply]
Yeah, I agree. I think we could have something like Stripe in the future where users can embed it entirely or an API (and custom elements) to build their own UI.
